### PR TITLE
Add debug API endpoints for batch, singleton, and debounce insights

### PR DESCRIPTION
**Pull request by BrunoScheufler**
## Description

This PR adds three new debug API endpoints to provide visibility into the state of batching, singleton locks, and debounce operations:

1. **GetBatchInfo** - Retrieves information about the current batch for a function and batch key, including:
   - Current batch ID
   - Number of items in the batch
   - Full event data for each item in the batch
   - Batch status

2. **GetSingletonInfo** - Retrieves the current singleton lock status for a given key, including:
   - Whether a lock is currently held
   - The run ID holding the lock (if any)

3. **GetDebounceInfo** - Retrieves information about the currently debounced event, including:
   - Whether a debounce is pending
   - The debounce ID and event ID
   - Full event data
   - Timeout and metadata (account, workspace, function IDs)

The implementation includes:
- New gRPC service definitions in proto files (batch.proto, singleton.proto, debounce.proto)
- Go implementations for each endpoint with proper error handling and validation
- Comprehensive test coverage for all three endpoints
- Integration with existing Redis clients for batch, singleton, and debounce state management
- Proper dependency injection through the debug API service options

## Motivation

These endpoints enable better debugging and monitoring of Inngest's execution features by allowing developers and operators to inspect the current state of batches, singleton locks, and debounce operations without direct Redis access. This is particularly useful for:
- Troubleshooting batch processing issues
- Verifying singleton lock behavior
- Inspecting pending debounce events
- Understanding why certain events may not be executing as expected

## Type of change

- [x] New feature (non-breaking change that adds functionality)

## Checklist

- [x] I've tested my own changes (comprehensive test coverage included)
- [x] Tests cover happy paths, error cases, and edge cases (nil clients, invalid UUIDs, missing data)

https://claude.ai/code/session_01WLa7iFMUvwZYP68Yb5wsz2

### CODE REVIEW

File: pkg/debugapi/batch.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	// Get the batch pointer key
+	batchKey := req.BatchKey
+	if batchKey == "" {
+		batchKey = "default"
+	}
+
+	// Hash the batch key the same way the batch manager does
+	hashedBatchKey := sha256.Sum256([]byte(batchKey))
```
Comment (BrunoScheufler) [inline pkg/debugapi/batch.go]: Please use the same function the batch manager uses

**Review by darwin67 — APPROVED** [review]
